### PR TITLE
Additional patch for CPython/3.8.12, duplicate libffi symbols

### PR DESCRIPTION
**Pull request by cubanpit**
I had issues compiling cpython/3.8.12 on Windows with shared libffi, the linker complained about duplicated symbols.
This patch solved the problem and I later found [this recent commit](https://github.com/python/cpython/commit/38f331d4656394ae0f425568e26790ace778e076#diff-6f6b7f83e2fb49775efdfa41b4aa4f8fadcf71f43c4f3bcf9f37743acafd3fdf) in the CPython repo that seems to apply the same change.

### CODE REVIEW

File: recipes/cpython/all/conandata.yml [hunk 1]
```diff
@@ -50,6 +50,8 @@ patches:
       base_path: "source_subfolder"
     - patch_file: "patches/3.8.12-0004-disable-macos-tcltk.patch"
       base_path: "source_subfolder"
+    - patch_file: "patches/3.8.12-0005-_ctypes-proj.patch"
+      base_path: "source_subfolder"
```
Comment (uilianries) [inline recipes/cpython/all/conandata.yml:54]:
```suggestion
      base_path: "source_subfolder"
```
Could you please add extra information, like `patch_description` and `patch_type`. Also, if it's part of upstream, please, point it on `patch_source`.

More information: https://github.com/conan-io/conan-center-index/blob/master/docs/conandata_yml_format.md#patches-fields

**Comment by cubanpit** [comment]
This is the message that pops up if I try to compile cpython/3.8.12 and shared libffi/3.4.2 without my patch:
```
malloc_closure.obj : error LNK2005: ffi_closure_alloc already defined in libffi.lib(libffi.dll) [C:\Users\pietrof\.conan\data\cpython\3.8.12\quantumatk\stable\build\656430ffe555a9fc989dd95dcd41975381d1186c\source_subfolder\PCbui
ld\_ctypes.vcxproj]
malloc_closure.obj : error LNK2005: ffi_closure_free already defined in libffi.lib(libffi.dll) [C:\Users\pietrof\.conan\data\cpython\3.8.12\quantumatk\stable\build\656430ffe555a9fc989dd95dcd41975381d1186c\source_subfolder\PCbuil
d\_ctypes.vcxproj]
C:\Users\pietrof\.conan\data\cpython\3.8.12\quantumatk\stable\build\656430ffe555a9fc989dd95dcd41975381d1186c\source_subfolder\PCbuild\amd64\_ctypes.pyd : fatal error LNK1169: one or more multiply defined symbols found [C:\Users\
pietrof\.conan\data\cpython\3.8.12\quantumatk\stable\build\656430ffe555a9fc989dd95dcd41975381d1186c\source_subfolder\PCbuild\_ctypes.vcxproj]
```
There is not issue about this opened on other projects, as far as I know.

**Comment by cubanpit** [comment]
Here I attach the entire log from building the package with my conanfile (sorry, not very minimal, I can produce a more minimal example if needed), and then the settings and the profile files.

[settings.yml.txt](https://github.com/conan-io/conan-center-index/files/9716294/settings.yml.txt)
[stdout.txt](https://github.com/conan-io/conan-center-index/files/9716295/stdout.txt)
[PROFILE_Windows_VS19md_if21_cygwin.txt](https://github.com/conan-io/conan-center-index/files/9716296/PROFILE_Windows_VS19md_if21_cygwin.txt)

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by uilianries** [comment]
@cubanpit Thank you for providing those logs.

**Comment by prince-chrismc** [comment]
Please open a PR upstream 🙏

**Review by prince-chrismc — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by ericLemanissier** [comment]
@cubanpit please close the PR for 10 second then reopen it so that it retriggers a build and finally merge this change

**Comment by uilianries** [comment]
@ericLemanissier still needs to added to access request list 😉

**Comment by cubanpit** [comment]
It seems that there is a build failure unrelated to my changes, any way to only build the specific Python version I am patching?

**Comment by uilianries** [comment]
@cubanpit I don't think is related to your PR, as your patch is only changing another version. It will need some investigation.

**Comment by prince-chrismc** [comment]
Did we recently update the mac images? https://stackoverflow.com/questions/63972113/big-sur-clang-invalid-version-error-due-to-macosx-deployment-target

**Comment by uilianries** [comment]
@prince-chrismc I don't think so. Pinging @danimtb and @davidsanfal Just in case.

**Comment by danimtb** [comment]
> @prince-chrismc I don't think so. Pinging @danimtb and @davidsanfal Just in case.

not really...

**Comment by prince-chrismc** [comment]
as a workaround id block that apple-clang 11 for that one version and see if that's enough

add a message pointing to the comment that highlighted the failure

**Pull request closed without merging**